code42/py42
# Chore/update device settings

### Description of Change ###

Adds the `IncydrDeviceSettings` class to manage device settings on Incydr devices, ie. computers without backup settings.  Previously, trying to get the device settings of an Incydr machine threw an error because we assume backup configuration values would be provided.

Created a whole new class instead of trying to reorganize the existing backup-related code to accommodate it.

- upgraded black to v22.3.0 which caused a lot of small changes
- upgraded CLA-assistant to latest version which will take affect on any future PRs 

### Testing Procedure ###

Users should be able to use the `DeviceService` methods `get_settings()` and `update_settings()` to view properties and update the _notes_ and _external_reference_ fields on the device settings of Incydr devices.

### PR Checklist ###
Did you remember to do the below?
